Setup fails with XSLT enabled

Following the [setup instructions](https://github.com/martinthomson/i-d-template/blob/master/doc/REPO.md) suggests just adding `USE_XSLT=true` to the setup command line, but this doesn't actually work.

> make: *** No rule to make target 'circle.yml', needed by 'setup-circle'.  Stop.

I think the addition of xsltproc to the Dockerfile actually obsoletes the setup-circle step entirely -- we should be able to just change the Makefile.

(Follow-up point -- if you *don't* use XSLT, the change to use a unified git cache is currently still keeping a cache of Julian's repo in the Dockerfile.  Does that matter?)

--- a/setup.mk
+++ b/setup.mk
@@ -52,16 +52,10 @@ setup-files: $(TEMPLATE_FILES) README.md
 	git add $^
 
 ifeq (true,$(USE_XSLT))
-setup-master: setup-makefile setup-circle
+setup-master: setup-makefile
 .PHONY: setup-makefile
 setup-makefile: Makefile
 	sed -i~ -e '1{h;s/^.*$$/USE_XSLT := true/;p;x}' $<
-	@-rm -f $<~
-	git add $<
-
-.PHONY: setup-circle
-setup-circle: circle.yml
-	sed -i~ -e '/^dependencies:/,/^  pre:$$/{p;s/^  pre:$$/    - sudo apt-get -qq update; sudo apt-get -q install xsltproc/;t;d}' $<
 	@-rm -f $<~
 	git add $<
 endif # USE_XSLT

--- a/tests/setup.feature
+++ b/tests/setup.feature
@@ -16,6 +16,22 @@ Feature: Initial setup
      and gitignore lists the xml file
      and a precommit hook is installed
 
+  Scenario:  Run setup script with XSLT on correctly-set-up directory
+    Given a git repo with a single Kramdown draft
+     when the setup script is run with "USE_XSLT=true"
+     then it succeeds
+     and a file is created called "Makefile" which contains "USE_XSLT := true"
+     and a file is created called "README.md"
+     and a file is created called "CONTRIBUTING.md"
+     and a file is created called "LICENSE.md"
+     and a file is created called ".gitignore"
+     and a file is created called ".travis.yml"
+     and a file is created called ".circleci/config.yml"
+     and a branch is created called "gh-pages" containing "index.html"
+     and a branch is created called "gh-pages" containing "issues.json"
+     and gitignore lists the xml file
+     and a precommit hook is installed
+
   Scenario:  Run setup script on directory with no draft
     Given an empty git repo
       and lib is cloned in

--- a/tests/steps/build_commands.py
+++ b/tests/steps/build_commands.py
@@ -41,6 +41,11 @@ def step_impl(context):
     run_with_capture(context, ["make", "-f", "lib/setup.mk"])
 
 
+@when(u'the setup script is run with "{option}"')
+def step_impl(context,option):
+    run_with_capture(context, ["make", "-f", "lib/setup.mk", option])
+
+
 @when(u'make is run')
 def step_impl(context):
     run_with_capture(context, ["make"])

--- a/tests/steps/results.py
+++ b/tests/steps/results.py
@@ -72,6 +72,16 @@ def step_impl(context):
             assert html_file in ghpages_files
 
 
+@then(u'a file is created called "{filename}" which contains "{text}"')
+def step_impl(context, filename, text):
+    context.execute_steps(
+        u'then a branch is created called "master" containing "%s"' % filename)
+    with cd(context.working_dir):
+        content = check_output(
+            ["git", "show", "master:%s" % filename]).decode("utf-8")
+        assert text in content
+
+
 @then(u'a file is created called "{filename}"')
 def step_impl(context, filename):
     context.execute_steps(

